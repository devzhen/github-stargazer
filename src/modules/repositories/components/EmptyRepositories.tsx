import * as R from 'ramda';
import React from 'react';

import Message from '@components/Message';

interface EmptyRepositoriesInterface {
  repositories: any;
  isFetchingNow: boolean;
}

const EmptyRepositories = ({
  repositories,
  isFetchingNow,
}: EmptyRepositoriesInterface) => {
  if (isFetchingNow || !R.isEmpty(repositories)) {
    return null;
  }

  return <Message message="User doesn't have repositories." offsetTop={30} />;
};

export default EmptyRepositories;
