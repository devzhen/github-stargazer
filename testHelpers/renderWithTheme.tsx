import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import 'jest-styled-components';

import theme from '@theme/index';

const renderWithTheme = (component: any) =>
  renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export default renderWithTheme;
