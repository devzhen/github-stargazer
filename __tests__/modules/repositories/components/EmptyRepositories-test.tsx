import React from 'react';
import 'jest-styled-components';

import EmptyRepositories from '@modules/repositories/components/EmptyRepositories';

// @ts-ignore
import renderWithTheme from '@testHelpers/renderWithTheme';

describe('EmptyRepositories repository component', () => {
  it('renders correctly, 1 case', () => {
    const tree = renderWithTheme(
      <EmptyRepositories isFetchingNow={false} repositories={[]} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 2 case', () => {
    const tree = renderWithTheme(
      <EmptyRepositories isFetchingNow={true} repositories={[]} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 3 case', () => {
    const tree = renderWithTheme(
      <EmptyRepositories isFetchingNow={false} repositories={[{ id: 1 }]} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly, 4 case', () => {
    const tree = renderWithTheme(
      <EmptyRepositories isFetchingNow={true} repositories={[{ id: 1 }]} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
