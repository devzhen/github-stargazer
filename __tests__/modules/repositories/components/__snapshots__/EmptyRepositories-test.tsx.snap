// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyRepositories repository component renders correctly, 1 case 1`] = `
<View
  offsetBottom={0}
  offsetTop={30}
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 30,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      },
    ]
  }
>
  <Text
    alignment="center"
    color="#eb9e38"
    onPress={[Function]}
    size={14}
    style={
      Array [
        Object {
          "color": "#eb9e38",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    weight="bold"
  >
    User doesn't have repositories.
  </Text>
</View>
`;

exports[`EmptyRepositories repository component renders correctly, 2 case 1`] = `null`;

exports[`EmptyRepositories repository component renders correctly, 3 case 1`] = `null`;

exports[`EmptyRepositories repository component renders correctly, 4 case 1`] = `null`;
